src/Property/Component/CompositeComponent.php

```diff
@@ -48,18 +66,33 @@ public function set($property_name, $value)
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function values()
```

Comment:
Values and prepareRender have duplicate code. Create a third private method with an argument for method and have the two public methods call that.